storage/range.go

```diff
@@ -769,21 +771,24 @@ func (r *Range) processRaftCommand(idKey cmdIDKey, index uint64, raftCmd proto.I
 
 	args := raftCmd.Cmd.GetValue().(proto.Request)
 	var reply proto.Response
+	var ctx context.Context
 	if cmd != nil {
 		// We initiated this command, so use the caller-supplied reply.
 		reply = cmd.Reply
+		ctx = cmd.ctx
 	} else {
 		// This command originated elsewhere so we must create a new reply buffer.
 		reply = args.CreateReply()
+		ctx = r.context()
 	}
 
-	err := r.applyRaftCommand(index, proto.RaftNodeID(raftCmd.OriginNodeID), args, reply)
+	err := r.applyRaftCommand(ctx, index, proto.RaftNodeID(raftCmd.OriginNodeID), args, reply)
```

Comment:
sounds fine Pete. I agree with that.

On Fri, Jun 12, 2015 at 10:06 AM Peter Mattis <notifications@github.com>
wrote:

> In storage/range.go
> <https://github.com/cockroachdb/cockroach/pull/1358#discussion_r32318394>:
>
> >  	}
> >
> > -	err := r.applyRaftCommand(index, proto.RaftNodeID(raftCmd.OriginNodeID), args, reply)
> > +	err := r.applyRaftCommand(ctx, index, proto.RaftNodeID(raftCmd.OriginNodeID), args, reply)
>
> I massively prefer to avoid doing such code movement in a PR like this
> which is mostly mechanical edits. Right now it is simple to see that Tobias
> was just adding a context parameter to lots of calls. Move the code around
> and the review becomes much more difficult.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/cockroachdb/cockroach/pull/1358/files#r32318394>.
>